Add generic rule for mass nouns ending in '-ware'

benvdh opened the pull request:
Fixes #119 .

SenseException commented on lib/Doctrine/Inflector/Rules/English/Uninflected.php:
```diff
@@ -31,6 +31,7 @@ public static function getPlural() : iterable
         yield new Pattern('middleware');
         yield new Pattern('people');
         yield new Pattern('software');
+        yield new Pattern('hardware');
```
I wonder: Does every word ending with ware has the same pluralization? We could cover all those words with a regex pattern.

garrettw replied:
I do believe so, except for the word "ware" itself which pluralizes to "wares".

SenseException replied:
This can be covered with a regex.

benvdh replied:
As long as we're dealing with nouns that seems to be case. As ware itself is a mass noun, and mass nouns by definition indicate a collection of multiple things. Hence, they have no separate plural form. See: https://en.oxforddictionaries.com/definition/ware#h47396550536000

The only edge case I could think of where a pattern could become tricky is when you go beyond nouns, and get names ending in -aware. Though such words would mostly be used as an adjective, rather than a noun.

@SenseException: Given these observations, would you like me to create a regex version instead?

garrettw replied:
Interesting, so it is unclear whether "ware" is being used as a mass noun or a singular noun from code's perspective -- so should we then pluralize it or not?

benvdh replied:
What would you think of the following regex?
`[A-Za-z]+ware$`

It excludes `ware` as such, allowing for the pluralisation to `wares` (which is mentioned in the Oxford dictionary example 1.2 of ware^1, but it would would exclude terms such as `hardware` or `software` from being pluralized as `hardwares` or `softwares`...

benvdh commented:
@SenseException @garrettw I have replaced it with a regex now, and added some test cases. Given that the testcases pass, the rule must be working. However, Scrutinizer seems to disagree that the line is being hit by the tests. Any advice on what I need to do to make scrutinizer pass?

benvdh commented:
> It seems to be a Scrutinizer thing, that went wrong. I guess your changes should be fine.

Thanks for the approval! Who will merge this? as I do not have write permissions to the repo...

jwage commented:
Thanks!
